### DISCUSSION TITLE
esModuleInterop in tsconfig.json

### DISCUSSION THREAD

**Discussion by r3m0t** [microsoft/vscode-discussions#149] [Extension Development QnA] [answered]
Hi, I noticed none of Microsoft's examples use this setting in their tsconfig.json even though the typescript documentation says it's a recommended setting. Has anybody got any working examples with it on, and are there any actual advantages/disadvantages to changing it from the examples' value of `false`?

**Comment by burkeholland** [comment]
Is this still an issue? Have you discovered yourself if this setting makes a difference?

**Reply by r3m0t** [reply]
Yeah, I get issues with default imports when it's on. No idea if it's the webpack config or what, but I have left it false. I just wish I had it on from the start...

**Reply by r3m0t** [reply]
Specifically, VS Code auto suggests default imports, but I have to change them to `import * as` to get them working.

**Reply by isidorn** [reply]
@mjbvz might be able to help here

**Comment by mjbvz** [comment, marked as answer]
This is a better question for either the TypeScript team or (if you have a specific problem) stack overflow

The option itself is documented [here](https://www.typescriptlang.org/tsconfig#esModuleInterop). We don't enable it in our repos/samples because we haven't needed to
